connect/connect.go

```diff
@@ -0,0 +1,61 @@
+// Package connect provides a Connector which allows to connect method
+// definition with their implementations in plugins
+package connect
```

Comment:
Can the package be called connector? then it's not `connect.connect(...)` but `connector.connect(...)`

Reply:
There is a `Connector` in `connect`, like there is a `Router` in `route`, so I think this is in line with how we name things. Only difference is, that there is a `Connect` method (which needs to be there for technical reasons required by how Go handles Generics - happy to explain live) whereas `route` does not need a `Route` method.